screen/edit-transaction/src/main/java/com/ivy/transaction/EditTransactionViewModel.kt

```diff
@@ -706,7 +709,10 @@ class EditTransactionViewModel @Inject constructor(
                     accountsChanged = false
                 }
 
-                transactionWriter.save(loadedTransaction().toEntity())
+                val trans = with(transactionMapper) {
+                    loadedTransaction().toEntity().toDomain()
+                }
+                trans.getOrNull()?.let { transactionWriter.save(it) }
```

Comment:
`TransactionWriter` is a legacy concept and must be deprecated. Use `TransactionRepository`

Reply:
Ahh, just rename the val

Reply:
Sure, here transactionWriter is just the parameter name, well i will change that